### PR TITLE
Detection 'area' element under cursor fixed (close #143)

### LINKED ISSUE #143: TestCafe can't reproduce a click on the 'area' element in Firefox
T299665

### PR DESCRIPTION
**Pull request by helen-dikareva**
/cc @inikulin @AlexanderMoskovkin

### CODE REVIEW

File: src/client/runner/automation/util.js [hunk 1]
```diff
@@ -190,17 +190,27 @@ export function getDragEndPoint (startPosition, to, currentDocument) {
 export function getElementUnderCursor (x, y, currentDocument, target) {
     var topElement = cursor.getElementUnderCursor(x, y, currentDocument);
 
-    if (!target || !domUtils.isDomElement(target) || topElement === target ||
-        (target.tagName.toLowerCase() !== 'a' && !$(target).parents('a').length) || !$(target).text().length)
+    if (!target || !domUtils.isDomElement(target) || topElement === target || !topElement)
         return topElement;
+
+    //NOTE: T299665 - Incorrect click on image with associated map element in Mozilla
+    if (browserUtils.isFirefox && target.tagName.toLowerCase() === 'area' &&
```
Comment (inikulin) [inline src/client/runner/automation/util.js]: `$(target)` is used 4 times. Let's  cache it:
```js
var $target = $(target);
```

**Comment by inikulin** [comment]
\r-

**Comment by helen-dikareva** [comment]
FPR

**Comment by inikulin** [comment]
lgtm

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Pull request merged**